### PR TITLE
Update publish-an-extension.md

### PR DESCRIPTION
**Pull request by thomaspaulmann**
## Description

Added a paragraph about sparse checkouts to improve the speed of git operations. The solution is based on [this discussion](https://raycastcommunity.slack.com/archives/C02HEMAF2SJ/p1654718297431099?thread_ts=1654716621.483799&cid=C02HEMAF2SJ) in the community.

## Checklist

- [ ] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [ ] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ ] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [ ] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — APPROVED** [review]

**Comment by luknl** [comment]
Just tried it and it worked, but there is a missing step that I think some people will miss. After cloning, the folder will be empty, so you need to create an /extensions folder at the root, and add your extension folder in there (ex: emoji-search).
What do you think?

**Comment by luknl** [comment]
> The extensions repository is big and to keep your git operations fast locally

Maybe be more specific with: "... and to avoid pulling all its content"? 

> This is especially useful when iterating on an existing extension.

I agree but it's useful also when you create an extension, so maybe rephrase it to "This is also useful"?
